### ISSUE TITLE
AS923 doesn't able to join any nearby hotspot

### ISSUE THREAD

**Issue by ubiru** [0hotpotman0/WioTerminal-LoRaWAN-Gateway-Tester#2]
Have you tested the as923 freq? And is it working?

i just received the wio terminal with the Lora/GNSS module a few days ago from seeedstudio and tested your version of code and also the original code from disk91. Both codes doesn't able to join nearby AS923 helium hotspots.

I've also informed this to disk91 and he would also test the code from his side.

So i don't know if there's something wrong in the code or maybe the wioterminal set that i received. But the configuration for the helium console and the input of all the keys and freq should have been correct. FYI, i'm using AS923_2.

**Comment by ubiru** [comment]
Hi 0hotpotman0,
Which AS923 region that you tested? AS923_1, AS923_2, AS023_3, or all of them?
Btw, how can i check my gateway segment network? The hotspots are sensecap, and in the sensecap dashboard it shows AS923_2 (for Indonesia, which is correct). There are 2 hotspots and those 2 hotspots are online, no relayed at all.
Furthermore, in wioterminal, also there's no selection for which AS923 region, just AS923. So that's why i was thinking that the problem of can't connect maybe due to the different AS923 (1,2, or 3).

**Comment by 0hotpotman0** [comment]
I have tested AS923_1 before, I thought 3 of them are pretty much the same, I will test others.
It will take time, you may try other regions before I test.
In further, We will be using the disk91 program as the main for helium, we have cooperation with disk91, this project is just for the TTN platform in future.

**Comment by ubiru** [comment]
No, the freq for every region in AS923 is different. So if you limit the freq to AS923_1 to the device, then it won't work in AS923_2 countries.
Here is the freq of uplink and downlink used in different AS923 https://wiki.dragino.com/index.php?title=End_Device_Frequency_Band

Since RP002-1.0.1LoRaWAN Regional Parameters, AS923 include three different default settings for different countries: AS923-1, AS923-2 and AS923-3.

AS923-1 use the same default uplink frequency as previous AS923 firmware. AS923-2 and AS923-3 are new with different default uplink frequency. User need to know exactly the frequency bands from the LoRaWAN server and choose the right firmware version for the end node.




AS920-923 & AS923-925 : AS923-1 (AS923)
Default Uplink channel:
923.2 - SF7BW125 to SF10BW125
923.4 - SF7BW125 to SF10BW125

Additional Uplink Channel: These channels are set during OTAA Join process from server. Below are the settings from TTN
AS920~AS923:
922.2 - SF7BW125 to SF10BW125
922.4 - SF7BW125 to SF10BW125
922.6 - SF7BW125 to SF10BW125
922.8 - SF7BW125 to SF10BW125
923.0 - SF7BW125 to SF10BW125
922.0 - SF7BW125 to SF10BW125
AS923 ~ AS925:
923.6 - SF7BW125 to SF10BW125
923.8 - SF7BW125 to SF10BW125
924.0 - SF7BW125 to SF10BW125
924.2 - SF7BW125 to SF10BW125
924.4 - SF7BW125 to SF10BW125
924.6 - SF7BW125 to SF10BW125
Downlink:
Uplink channels 1-8 (RX1)
923.2 - SF10BW125 (RX2)


AS923-2
Default Uplink channel:
921.4 - SF7BW125 to SF10BW125
921.6 - SF7BW125 to SF10BW125

Additional Uplink Channels: These channels are set during OTAA Join process from server.
Downlink:
Uplink channels 1-8 (RX1)
921.4 - SF10BW125 (RX2)


AS923-3
Default Uplink channel:
916.6 - SF7BW125 to SF10BW125
916.8 - SF7BW125 to SF10BW125

Additional Uplink Channels: These channels are set during OTAA Join process from server.
Downlink:
Uplink channels 1-8 (RX1)
916.6 - SF10BW125 (RX2)
